app/scripts/controllers/network/provider-api-tests/helpers.ts

```diff
@@ -52,96 +51,89 @@ function debug(...args) {
+};
+type ResponseBody = { body: JSONRPCResponse };
+type BodyOrResponse = ResponseBody | Response;
+type CurriedMockRpcCallOptions = {
+  request: Request;
+  // The response data.
+  response?: BodyOrResponse;
+  /**
+   * An error to throw while making the request.
+   * Takes precedence over `response`.
+   */
+  error?: Error | string;
+  /**
+   * The amount of time that should pass before the
+   * request resolves with the response.
+   */
+  delay?: number;
+  /**
+   * The number of times that the request is
+   * expected to be made.
+   */
+  times?: number;
+};
+
+type MockRpcCallOptions = {
+  // A nock scope (a set of mocked requests scoped to a certain base URL).
+  nockScope: nock.Scope;
+} & CurriedMockRpcCallOptions;
+
+type MockRpcCallResult = nock.Interceptor | nock.Scope;
 
 /**
  * Mocks a JSON-RPC request sent to the provider with the given response.
  * Provider type is inferred from the base url set on the nockScope.
  *
- * @param {MockRpcCallOptions} args - The arguments.
- * @param {NockScope} args.nockScope - A nock scope (a set of mocked requests
- * scoped to a certain base URL).
- * @param {object} args.request - The request data.
- * @param {{body: string} | {httpStatus?: number; id?: number; method?: string; params?: string[]}} [args.response] - Information
- * concerning the response that the request should have. If a `body` property is
- * present, this is taken as the complete response body. If an `httpStatus`
- * property is present, then it is taken as the HTTP status code to respond
- * with. Properties other than these two are used to build a complete response
- * body (including `id` and `jsonrpc` properties).
- * @param {Error | string} [args.error] - An error to throw while making the
- * request. Takes precedence over `response`.
- * @param {number} [args.delay] - The amount of time that should pass before the
- * request resolves with the response.
- * @param {number} [args.times] - The number of times that the request is
- * expected to be made.
- * @returns {NockScope} The nock scope.
+ * @param args - The arguments.
+ * @param args.nockScope - A nock scope (a set of mocked requests scoped to a
+ * certain base URL).
+ * @param args.request - The request data.
+ * @param args.response - Information concerning the response that the request
+ * should have. If a `body` property is present, this is taken as the complete
+ * response body. If an `httpStatus` property is present, then it is taken as
+ * the HTTP status code to respond with. Properties other than these two are
+ * used to build a complete response body (including `id` and `jsonrpc`
+ * properties).
+ * @param args.error - An error to throw while making the request. Takes
+ * precedence over `response`.
+ * @param args.delay - The amount of time that should pass before the request
+ * resolves with the response.
+ * @param args.times - The number of times that the request is expected to be
+ * made.
+ * @returns The nock scope.
```

Comment:
Nit: For some reason, this returns the _interceptor_ in some cases (if no error or response is specified).

While this mismatch between the docs and the method is pre-existing, I noticed it now because it explained why we needed to check for `persist` in the `mockAllBlockTrackerRequests` function.

Reply:
Yup, you're totally right. I completely missed this when I was implementing it for the first time here, and the mistake got ported over to the core side. We can definitely clean this up post-merge.